pkg/es/config/flags.go

```diff
@@ -70,342 +68,283 @@ const (
 			"ILM policy must be manually created in ES before startup. Supported only for elasticsearch version 7+.")
 	flagSet.Bool(
-		nsConfig.namespace+suffixCreateIndexTemplate,
-		nsConfig.CreateIndexTemplates,
+		namespace+suffixCreateIndexTemplate,
+		config.CreateIndexTemplates,
 		"Create index templates at application startup. Set to false when templates are installed manually.")
 	flagSet.Uint(
-		nsConfig.namespace+suffixVersion,
+		namespace+suffixVersion,
 		0,
 		"The major Elasticsearch version. If not specified, the value will be auto-detected from Elasticsearch.")
 	flagSet.Bool(
-		nsConfig.namespace+suffixSnifferTLSEnabled,
-		nsConfig.Sniffing.UseHTTPS,
+		namespace+suffixSnifferTLSEnabled,
+		config.Sniffing.UseHTTPS,
 		"Option to enable TLS when sniffing an Elasticsearch Cluster ; client uses sniffing process to find all nodes automatically, disabled by default")
 	flagSet.Int(
-		nsConfig.namespace+suffixMaxDocCount,
-		nsConfig.MaxDocCount,
+		namespace+suffixMaxDocCount,
+		config.MaxDocCount,
 		"The maximum document count to return from an Elasticsearch query. This will also apply to aggregations.")
 	flagSet.String(
-		nsConfig.namespace+suffixLogLevel,
-		nsConfig.LogLevel,
+		namespace+suffixLogLevel,
+		config.LogLevel,
 		"The Elasticsearch client log-level. Valid levels: [debug, info, error]")
 	flagSet.String(
-		nsConfig.namespace+suffixSendGetBodyAs,
-		nsConfig.SendGetBodyAs,
+		namespace+suffixSendGetBodyAs,
+		config.SendGetBodyAs,
 		"HTTP verb for requests that contain a body [GET, POST].")
 	flagSet.Duration(
-		nsConfig.namespace+suffixAdaptiveSamplingLookback,
-		nsConfig.AdaptiveSamplingLookback,
+		namespace+suffixAdaptiveSamplingLookback,
+		config.AdaptiveSamplingLookback,
 		"How far back to look for the latest adaptive sampling probabilities")
-	if nsConfig.namespace == archiveNamespace {
+	if namespace == archiveNamespace {
 		flagSet.Bool(
-			nsConfig.namespace+suffixEnabled,
-			nsConfig.Enabled,
+			namespace+suffixEnabled,
+			config.Enabled,
 			"Enable extra storage")
 	} else {
 		// MaxSpanAge is only relevant when searching for unarchived traces.
 		// Archived traces are searched with no look-back limit.
 		flagSet.Duration(
-			nsConfig.namespace+suffixMaxSpanAge,
-			nsConfig.MaxSpanAge,
+			namespace+suffixMaxSpanAge,
+			config.MaxSpanAge,
 			"The maximum lookback for spans in Elasticsearch")
 	}
-	nsConfig.getTLSFlagsConfig().AddFlags(flagSet)
+	tlsFlagsCfg := tlscfg.ClientFlagsConfig{
+		Prefix: namespace,
+	}
+	tlsFlagsCfg.AddFlags(flagSet)
 }
 
 // InitFromViper initializes Options with properties from viper
-func (opt *Options) InitFromViper(v *viper.Viper) {
-	initFromViper(&opt.Primary, v)
-	for _, cfg := range opt.others {
-		initFromViper(cfg, v)
+func (c *Configuration) InitFromViper(v *viper.Viper) {
+	if c.IsArchive {
+		initFromViper(c, v, archiveNamespace)
+		return
```

Comment:
@yurishkuro thoughts on how we should be initializing the configuration struct now that we're only holding one config?

Reply:
well, you can't really make clear judgement if you only change the configs, because it really depends on how the callers use the factory, e.g. queryservice/main